noir-projects/noir-contracts/contracts/private_fpc_contract/src/main.nr

```diff
@@ -0,0 +1,35 @@
+contract PrivateFPC {
+    use dep::aztec::protocol_types::{abis::function_selector::FunctionSelector, address::AztecAddress, traits::is_empty};
+    use dep::aztec::state_vars::SharedImmutable;
+    use dep::private_token::PrivateToken;
+    use dep::aztec::context::gas::GasOpts;
+
+    #[aztec(storage)]
+    struct Storage {
+        other_asset: SharedImmutable<AztecAddress>,
+        admin_npk_m_hash: SharedImmutable<Field>
+    }
+
+    #[aztec(public)]
+    #[aztec(initializer)]
+    fn constructor(other_asset: AztecAddress, admin_npk_m_hash: Field) {
+        storage.other_asset.initialize(other_asset);
+        storage.admin_npk_m_hash.initialize(admin_npk_m_hash);
+    }
+
+    #[aztec(private)]
+    fn fund_transaction_privately(amount: Field, asset: AztecAddress, nonce: Field) {
+        assert(asset == storage.other_asset.read_private());
+        // convince the FPC we are not cheating
+        context.push_new_nullifier(nonce, 0);
+        // allow the FPC to reconstruct their fee note
+        context.emit_unencrypted_log(nonce);
```

Comment:
Lol at `dangerously...` We could also emit it as encrypted, but since the FPC is paying for this, and I don't think they care if nonce is revealed, they would prefer it to be unencrypted.

Reply:
Name's inspired by React's https://legacy.reactjs.org/docs/dom-elements.html#dangerouslysetinnerhtml